system-tests/lib/cre/contracts/contracts.go

```diff
@@ -195,6 +198,7 @@ func ConfigureKeystone(input cre.ConfigureKeystoneInput, capabilityRegistryConfi
 		}
 
 		dons.c[donName] = donConfig{
+			id:              uint32(donMetadata.ID), //nolint:gosec // G115
```

Comment:
you could also use `libc.MustSafeUint32FromUint64(donMetadata.ID` where libc is `libc "github.com/smartcontractkit/chainlink/system-tests/lib/conversions"`